pkg/controller/migmigration/backup.go

```diff
@@ -358,6 +443,93 @@ func (t *Task) annotateStorageResources() error {
 		}
 	}
 
+	return nil, resticPodCount
+}
+
+func (t *Task) areStagePodsCreated() (bool, error) {
+	client, err := t.getSourceClient()
+	if err != nil {
+		return false, err
+	}
+	// check if we have already created dummy pods
+	// if so, return true
+	uniqueBackupLabelKey := fmt.Sprintf("%s-%s", pvBackupLabelKey, t.PlanResources.MigPlan.UID)
+	labelSelector := map[string]string{
+		fmt.Sprintf("%s-copy", uniqueBackupLabelKey): "true",
+	}
+	options := k8sclient.MatchingLabels(labelSelector)
+	podList := corev1.PodList{}
+	err = client.List(context.TODO(), options, &podList)
+	if err != nil {
+		return false, err
+	}
+	podCount := len(podList.Items)
+	if podCount == 0 {
+		return false, nil
+	}
+	readyCount := 0
+	for _, pod := range podList.Items {
+		if pod.Status.Phase == corev1.PodRunning {
+			readyCount += 1
+		}
+	}
+	// Check if all pods are ready
+	if readyCount == podCount {
```

Comment:
This function doesn't have a **specific number of stage pods** its trying to verify the existence of, it's just querying for the fact that all existing Pods are Ready. 

@dymurray do you see any opportunity to failure here if we were only partially successful in creating Pods? Guessing we wouldn't get past the previous step of creating the stage pods in that case.

Reply:
I have updated the function to also verify it has created the same number of stage pods as restic annotations we have found. And it is possible that we could fail here if the pods aren't successful in coming up, but if that's the case then the migration cannot continue if the PVs are set to copy so I think failing is the right thing to do